Neuron/Services/Common/Token.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  Token.swift
+//  Neuron
+//
+//  Created by 晨风 on 2018/11/20.
+//  Copyright © 2018 Cryptape. All rights reserved.
+//
+
+import UIKit
+import Web3swift
+import PromiseKit
+import BigInt
+import EthereumAddress
+
+class Token {
+    var name = ""
+    var iconUrl: String? = ""
+    var address = ""
+    var symbol = ""
+    var chainName: String? = ""
+    var chainId = ""
+    var chainHosts = ""
+    var isNativeToken = false
+    var walletAddress = ""
+
+    private(set) var balance: Double? {
+        didSet {
+            guard let realm = tokenModel.realm else { return }
```

Comment:
I propose an approach that removes potential problem with Realm issue:

* Don't hold the `tokenModel` instance, instead, save its primary key.
* When updating, just query the token model from a fresh new realm instance and write to it.
* But still need to consider: since the new `Token` model is passed by value, how to sync values if somewhere we already have a value, and its balance from another `Token` (thus the underlying `TokenModel` instance) changes.